### DISCUSSION TITLE
Anlernen in zwei verschiedenen Setups scheitert

### DISCUSSION THREAD

**Discussion by m0ppers** [OpenCCU/OpenCCU#2078] [🇩🇪 Fragen&Antworten]
Hallo,

ich versuche raspberrymatic via docker+USB auf einem helios64 ans Laufen zu bekommen und laufe in diverse Probleme und habe jetzt irgendwie keinen Hebel mehr.

Ich habe bisher nur ein EVO Heizkörperthermostat (wollte erstmal testen, bevor ich weiter mache) und bekomme es nicht hin.

Nachdem ich die pivccu module (auf dem helios64 läuft armbian) installiert hatte und den docker daemon angepasst hatte kann ich mit folgendem docker aufruf ein raspberrymatic starten, was zu funktionieren scheint:

`docker run --tty --interactive --name ccu -d           --volume ccu_data:/usr/local:rw --volume /lib/modules:/lib/modules:ro            --volume /run/udev/control:/run/udev/control            --publish 8080:80 --privileged --restart always --stop-timeout 30 --cpu-rt-runtime 950000 --ulimit rtprio=99           --hostname homematic-raspi ghcr.io/jens-maus/raspberrymatic:latest`

Die docker logs sehen jetzt sauber aus:

```
Starting watchdog...
Identifying host system: oci, OK
Initializing RTC Clock: onboard, OK
Running sysctl: OK
Checking for Factory Reset: not required
Checking for Backup Restore: not required
Initializing System: OK
Setup ca-certificates: OK
Starting logging: OK
Populating /dev using udev: done
Init onboard LEDs: init, OK
Starting irqbalance: OK
Starting iptables: OK
Starting network: eth0: link up, fixed, firewall, inet up, 172.17.0.2, OK
Identifying Homematic RF-Hardware: ....HmRF: HMIP-RFUSB/eQ-3 HmIP-RFUSB@usb-xhci-hcd.0.auto-1.2, HmIP: HMIP-RFUSB/eQ-3 HmIP-RFUSB@usb-xhci-hcd.0.auto-1.2, OK
Updating Homematic RF-Hardware: HMIP-RFUSB: 4.4.18, not necessary, OK
Starting hs485dLoader: disabled
Starting xinetd: OK
Starting eq3configd: OK
Starting lighttpd: creating new SSL cert... OK
Starting ser2net: disabled
Starting ssdpd: OK
Starting NUT services: disabled
Initializing Third-Party Addons: OK
Starting LGWFirmwareUpdate: ...OK
Setting LAN Gateway keys: OK
Starting hs485d: disabled
Starting multimacd: ..OK
Starting rfd: .OK
Starting HMIPServer: ..........OK
Starting ReGaHss: .OK
Starting CloudMatic: OK
Starting NeoServer: OK
Starting Third-Party Addons: OK
Starting crond: OK
Setup onboard LEDs: booted, OK
Finished Boot: 3.65.11.20221005 (raspmatic_oci_arm64)
```

Ich hatte zunächst mit dem multimacd zu kämpfen, aber das scheint jetzt alles sauber zu sein.

Das USB Modul liegt mit dem beigelieferten Kabel so weit weg, wie es halt kann - ich habe hier auch nur eine 60qm Wohnung zu versorgen und keine Villa - die Entfernungen sind also kurz.  In der Software taucht das Modul dann sauber im Eingang auf und scheint glücklich.
Wenn ich jetzt mein EVO Thermostat mit einem touch in den anlernmodus versetze und im Raspberrymatic bei HmIP auch den Anlernmodus starte, passiert genau nichts. Mir ist auch völlig unklar, wie man da irgendwie herausfinden kann was denn los ist. Das docker log ist ja extrem schweigsam. Ich sehe in `/var/log/messages` im container:

```
[...]
Dec  1 16:26:46 homematic-raspi daemon.err xinetd[979]: Unable to read included directory: /etc/config/xinetd.d [file=/etc/xinetd.conf] [line=14]
Dec  1 16:26:46 homematic-raspi daemon.crit xinetd[979]: 979 {init_services} no services. Exiting...
Dec  1 16:27:09 homematic-raspi local0.err ReGaHss: ERROR: original file not exists, so try load new file. [LoadOM():iseDOM.cpp:2610]
Dec  1 16:27:09 homematic-raspi local0.err ReGaHss: ERROR: new file not exists, so try load bak file. [LoadOM():iseDOM.cpp:2622]
Dec  1 16:27:09 homematic-raspi local0.err ReGaHss: ERROR: failed open file= /etc/config/homematic.regadom.bak [LoadFromFile():iseDOMpersist.cpp:78]
Dec  1 16:27:09 homematic-raspi local0.err ReGaHss: ERROR: (202) failed! Already in map! [Insert():iseDOMobj.cpp:526]
Dec  1 16:27:09 homematic-raspi local0.err ReGaHss: ERROR: IseAddFavorite failed! ID=202 [IseAddFavorite():iseDOM.cpp:644]
Dec  1 16:27:09 homematic-raspi local0.err ReGaHss: ERROR: (203) failed! Already in map! [Insert():iseDOMobj.cpp:526]
Dec  1 16:27:09 homematic-raspi local0.err ReGaHss: ERROR: IseAddFavorite failed! ID=203 [IseAddFavorite():iseDOM.cpp:644]
Dec  1 16:27:09 homematic-raspi local0.err ReGaHss: ERROR: (204) failed! Already in map! [Insert():iseDOMobj.cpp:526]
Dec  1 16:27:09 homematic-raspi local0.err ReGaHss: ERROR: IseAddFavorite failed! ID=204 [IseAddFavorite():iseDOM.cpp:644]
Dec  1 16:27:11 homematic-raspi user.info homematic: meine-homematic.de Versions - Upgrade auf 1.503
Dec  1 16:27:11 homematic-raspi user.info homematic: meine-homematic.de VPN Dienst und Autoupdate DEAKTIVIERT
Dec  1 15:27:12 homematic-raspi daemon.info : starting pid 1345, tty '': '/bin/mv /tmp/boot.log /var/log/boot.log'
Dec  1 15:27:12 homematic-raspi daemon.info : starting pid 1346, tty '/dev/null': '/usr/bin/monit -Ic /etc/monitrc'
Dec  1 16:27:12 homematic-raspi user.info monit[1346]:  New Monit id: d6a2c27cf4f7ae003526677f3876d0df  Stored in '/usr/local/etc/monit.id'
Dec  1 16:27:12 homematic-raspi user.info monit[1346]: Starting Monit 5.30.0 daemon with http interface at /var/run/monit.sock
Dec  1 16:27:12 homematic-raspi user.info monit[1346]: 'homematic-raspi' Monit 5.30.0 started
Dec  1 16:27:27 homematic-raspi user.err monit[1346]: 'sshdEnabled' status failed (1) -- no output
Dec  1 16:27:27 homematic-raspi user.err monit[1346]: 'hs485dEnabled' status failed (2) -- grep: /var/etc/hs485d.conf: No such file or directory
Dec  1 16:27:28 homematic-raspi user.err monit[1346]: 'hmlangwEnabled' status failed (1) -- no output
Dec  1 16:27:28 homematic-raspi user.warn monit[1346]: 'hasUSB' status failed (1) -- no output
Dec  1 16:27:28 homematic-raspi user.err monit[1346]: 'hb_rf_eth-CheckEnabled' status failed (1) -- no output
Dec  1 16:27:28 homematic-raspi user.warn monit[1346]: 'rpi4usb3Check' status failed (1) -- no output
Dec  1 16:27:28 homematic-raspi user.err monit[1346]: 'tailscaleEnabled' status failed (1) -- no output
Dec  1 16:27:28 homematic-raspi user.err monit[1346]: Lookup for '/media/usb1' filesystem failed  -- not found in /proc/self/mounts
Dec  1 16:27:28 homematic-raspi user.err monit[1346]: Filesystem '/media/usb1' not mounted
Dec  1 16:27:28 homematic-raspi user.err monit[1346]: 'usb1' unable to read filesystem '/media/usb1' state
Dec  1 16:27:28 homematic-raspi user.info monit[1346]: 'usb1' trying to restart
Dec  1 16:27:44 homematic-raspi user.warn monit[1346]: 'hasUSB' status failed (1) -- no output
Dec  1 16:27:44 homematic-raspi user.warn monit[1346]: 'rpi4usb3Check' status failed (1) -- no output
Dec  1 16:27:44 homematic-raspi user.err monit[1346]: Filesystem '/media/usb1' not mounted
Dec  1 16:27:44 homematic-raspi user.err monit[1346]: 'usb1' unable to read filesystem '/media/usb1' state
Dec  1 16:27:44 homematic-raspi user.info monit[1346]: 'usb1' trying to restart
Dec  1 16:28:00 homematic-raspi user.warn monit[1346]: 'hasUSB' status failed (1) -- no output
Dec  1 16:28:00 homematic-raspi user.err monit[1346]: 'rpi4usb3Check' status failed (1) -- no output
Dec  1 16:28:00 homematic-raspi user.err monit[1346]: Filesystem '/media/usb1' not mounted
Dec  1 16:28:00 homematic-raspi user.err monit[1346]: 'usb1' unable to read filesystem '/media/usb1' state
Dec  1 16:28:00 homematic-raspi user.info monit[1346]: 'usb1' trying to restart
Dec  1 16:28:00 homematic-raspi local0.warn ReGaHss: WARNING: XMLRPC 'getValue': rpcClient.isFault() failed (url: xmlrpc://127.0.0.1:32010, params: {"00041F299ECD99:0","CARRIER_SENSE_LEVEL"}, result: [faultCode:-5,faultString:"Unknown Parameter value for value key: CARRIER_SENSE_LEVEL"]) [CallXmlrpcMethod():iseXmlRpc.cpp:2608]
Dec  1 16:28:00 homematic-raspi local0.err ReGaHss: ERROR: XMLRPC 'getValue' call failed (interface: 1009, params: {"00041F299ECD99:0","CARRIER_SENSE_LEVEL"}) [CallGetValue():iseXmlRpc.cpp:1435]
Dec  1 16:28:00 homematic-raspi local0.err ReGaHss: ERROR: CallGetValue failed; sVal = 0.000000 [ReadValue():iseDOMdpHSS.cpp:124]
Dec  1 16:28:16 homematic-raspi user.warn monit[1346]: 'hasUSB' status failed (1) -- no output
Dec  1 16:28:16 homematic-raspi user.err monit[1346]: Filesystem '/media/usb1' not mounted
Dec  1 16:28:16 homematic-raspi user.err monit[1346]: 'usb1' unable to read filesystem '/media/usb1' state
Dec  1 16:28:16 homematic-raspi user.info monit[1346]: 'usb1' trying to restart
Dec  1 16:28:32 homematic-raspi user.err monit[1346]: 'hasUSB' status failed (1) -- no output
```

Das mit dem Carrier Sense Level sieht fragwürdig auch. Darüber finde ich auch einiges im Forum, aber ich weiss überhaupt nicht, nach was genau ich da ausschau halten soll. Dieser Thread hier hat mir ein paar Hinweise gegeben, aber geholfen hat da nichts: https://homematic-forum.de/forum/viewtopic.php?f=65&t=76473 .Wenn ich das USB Device auswähle steht dort Carrier Sense Level 0%. Und DC 0%. Im Docker Container komme ich leider auch nicht an die Zentralenwartung ran. Da bekomme ich lediglich eine ausgegraute Seite.

Kann man irgendwie wiresharken oder sowas mit dem USB modul? Kann man damit irgendwie ausserhalb der Software mal reden? Das ist halt eine komplette Blackbox :| Vielleicht habe ich ja wirklich irgendwas beim Löten verkackt. Aber andererseits taucht es auch in der Hilfe Seite inkl. einer Seriennummer auf. Das sieht ja erstmal gut aus?!

Da ich da nicht mehr weiter kam und gerade das Glück habe, dass ich Zugang zu einer zweiten Location habe, wo ähnliches probiert wird (geht auch noch nicht), habe ich meinen Kram mal dahin mitgenommen und es da probiert.

Auf dem dortigen RPi4 ein frisches raspberrymatic image installiert (also kein Docker oder sonstiges). Dort wird auch der USB stick sauber erkannt, aber findet dann keine Geräte: Weder die dortigen Geräte - Basic Thermostat und ein Wandthermostat - noch mein EVO. 

Dort gab es aber ausserdem noch das Funkmodul, was direkt auf den Pi montiert werden kann. Das habe ich dann auch mal probiert (vorher nochmal Image komplett frisch auf die SD Karte gepackt). Siehe da: Es findet mein EVO sofort. An den dortigen Geräten kann ich aber resetten und sonstwas tuen: Die werden nicht gefunden.

Wenn mit dem Pi Modul alles aufgetaucht wäre, wäre der Fall ja klar und ich hätte einfach beim Löten irgendwie subtil verkackt. Da dort aber wiederrum nur ein Bruchteil der Geräte erkannt wird - ja ich habe die dortigen THermostate komplett resettet -, habe ich jetzt keine Lust wieder sinnlos Geld auf das Problem zu werfen (neuen USB Stick probieren). Ich bin alles andere als ein Lötexperte, aber hin und wieder löte ich schon mal was. Die Lötpunkte sehen (für mich) sauber aus und wie gesagt im UI taucht alles richtig inkl. Seriennummer etc. auf.

Ich brauche irgendwie ein paar Hinweise, wie ich von hier aus weiter komme :| danke!

**Comment by ntdsutil** [comment]
Hey! das klingt doch sehr ähnlich zu meiner gestern gestarteten Disukussion hier: https://github.com/jens-maus/RaspberryMatic/discussions/2082

Ich habe einen anderen Aufbau aber die gleiche Problematik. Kannst du dir ja mal durchlesen.

Mittlerweile habe ich einen zweiten Stick zusammengebaut und immernoch das gleiche Problem.
Witzigerweise habe ich das gleiche Problem auch in neuen Proxmox Instanzen verifizieren können - mit BEIDEN Sticks.
Gut können auch beide kaputt sein aber ich dachte eher, dass es evtl. an meiner Proxxmox HW liegt also hab ich auf dem Windows Rechner meiner Frau eine VM mit VirtualBox aufgesetzt und habe dort genau das gleiche Problem.
Also sind meiner Meinung nach entweder beide Sticks kaputt oder es gibt irgendein Problem mit den RF-USB Sticks und RaspberryMatic.

Zum Thema CarrierSense: Ich hatte zu Beginn des Problems auch erstmalig CS Meldungen im LOG, davor lief das 3 Monate ohne ein Problem. Habe aber meine gesamte HW schon aus dem Technikraum in ein anderes Zimmer gepackt um das einen Einfluss hat. Leider nicht....

Zum Thema Wireshark/Logging: Ich habe mir mal ein SDR besorgt und will schauen ob ich was auf dem Kanal sehe aber wahrscheinlich wird mich das nicht wirklich weiterbringen. Du kannst den LogLevel anheben und dann wirst du die Nachrichten sehen. Vielleicht könntest du das mal machen und hier hochladen dann kann ich das mit meinen vergleichen...

Grüße
